### PR TITLE
Feat/24 and 27 Apollo setup and trading grid

### LINKED ISSUE #24: Trading app apollo setup

### LINKED ISSUE #27: Trading view page layout template
- Create page template for trading view
- Create tab functionality

### PR DESCRIPTION
**Pull request by mattrussell36**
Closes #24 
Closes #27

- Adds apollo setup
- Adds page layout styles for the trading page
- Adds simple mobile layout
- Adds tab functionality for some of the grid sections (orders/positions and trades/book). The state of these tabs are managed by query string values
- Adds a PageQueryContainer component to handle consistent error and loading states on a page level basis.

Large screens:
Screen Shot 2022-03-01 at 5 10 32 PM

Small screens:
mobile-trading

### CODE REVIEW

File: apps/trading/pages/markets/[marketId].page.tsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+  if (loading || !data) {
+    return <div>Loading...</div>;
+  }
+
+  return (
+    <>
+      {error ? (
+        <div>Something went wrong: {error.message}</div>
+      ) : (
```
Comment (dexturr) [inline apps/trading/pages/markets/[marketId].page.tsx:41]: (followup) non-blocking: would be nice to wrap this is a query component so we don't implement this a million times

File: apps/trading/pages/markets/[marketId].page.tsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        <div className="h-full max-h-full grid gap-[1px] bg-black grid-cols-[300px_2fr_1fr_1fr] grid-rows-[min-content_1fr_200px]">
+          <header className="bg-white col-span-4 p-8">
+            <h1>Market: {query.marketId}</h1>
+          </header>
+          <GridChild name="Ticket">Content</GridChild>
```
Comment (dexturr) [inline apps/trading/pages/markets/[marketId].page.tsx:46]: (question) non-blocking: Are we going i18n first or just english for first version?

**Review by dexturr — APPROVED** [review]
Looks good to me would like @bglownia to have a review of the nx specific stuff as this is newer to me

File: apps/trading/pages/markets/[marketId].page.tsx (hunk 2 above)
Comment (mattrussell36) [reply apps/trading/pages/markets/[marketId].page.tsx]: IMO lets leave i18n until we actually have some full translations ready to go. Open for debate though

File: apps/trading/pages/markets/[marketId].page.tsx (hunk 1 above)
Comment (mattrussell36) [reply apps/trading/pages/markets/[marketId].page.tsx]: Done! I've added a PageQueryContainer component which takes the query and query options as props, handles consistent loading and error states and passes the resulting data as a render prop.

I've also added it to the market list page.

File: apps/trading/pages/markets/trade-grid.tsx [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+          <div style={{ width, height }} className="overflow-auto">
+            {children}
+          </div>
+        )}
+      </AutoSizer>
+    </section>
+  );
+};
+
+type View = keyof typeof Views;
```
Comment (dexturr) [inline apps/trading/pages/markets/trade-grid.tsx:77]: chore (non-blocking): probably will be used by other windows, can we put this into apps/trading/pages/markets/trading-components.tsx

**Review by dexturr — CHANGES_REQUESTED** [review]
Couple of blocking things, but nothing major. LGTM mainly

File: apps/trading/pages/markets/[marketId].page.tsx [hunk 4]
```diff
@@ -0,0 +1,83 @@
+import { gql } from '@apollo/client';
+import { useRouter } from 'next/router';
+import React, { useEffect, useState } from 'react';
+import debounce from 'lodash/debounce';
```
[the remark this answers is not in the document]
Comment (dexturr) [reply apps/trading/pages/markets/[marketId].page.tsx:4]: Either this or the above comment

File: apps/trading/pages/markets/[marketId].page.tsx [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+  return (
+    <PageQueryContainer<Market, MarketVariables>
+      query={MARKET_QUERY}
+      options={{
+        variables: { marketId: query.marketId as string },
+        skip: !query.marketId,
+      }}
+    >
+      {(data) =>
```
Comment (sam-keen) [inline apps/trading/pages/markets/[marketId].page.tsx:37]: Tiny thing: could destructure `market`

File: apps/trading/pages/markets/[marketId].page.tsx [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+
+    // Something sensible for server rendered page
+    return {
+      w: 1200,
+      h: 900,
+    };
+  });
+
+  useEffect(() => {
+    const handleResize = debounce((event) => {
```
Comment (sam-keen) [inline apps/trading/pages/markets/[marketId].page.tsx:68]: Could also destructure `target`

File: apps/trading/pages/markets/index.page.tsx [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+    }
+  }
+`;
+
+const Markets = () => {
+  const { pathname } = useRouter();
+
+  return (
+    <PageQueryContainer<Markets> query={MARKETS_QUERY}>
+      {(data) => (
```
Comment (sam-keen) [inline apps/trading/pages/markets/index.page.tsx:20]: OMG I'm so much fun. Could destructure here too.

**Review by dexturr — APPROVED** [review]

**Pull request merged**